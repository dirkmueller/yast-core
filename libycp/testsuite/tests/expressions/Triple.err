Parsed:
----------------------------------------------------------------------
((size ([]) > 0) ? true : false == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(contains ([1, 2, 3], 5) ? "yes" : "no" == "no")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((size ([]) > 0) ? 1 : -1 == -1)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(1 == 1)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(2 == 2)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((0 < 0) ? (0 < 0) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((0 < 0) ? (0 < 1) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((0 < 0) ? (1 < 0) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((0 < 0) ? (1 < 1) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((0 < 1) ? (0 < 0) ? 1 : 2 : 3 == 2)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((0 < 1) ? (0 < 1) ? 1 : 2 : 3 == 1)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((0 < 1) ? (1 < 0) ? 1 : 2 : 3 == 2)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((0 < 1) ? (1 < 1) ? 1 : 2 : 3 == 2)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((1 < 0) ? (0 < 0) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((1 < 0) ? (0 < 1) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((1 < 0) ? (1 < 0) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((1 < 0) ? (1 < 1) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((1 < 1) ? (0 < 0) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((1 < 1) ? (0 < 1) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((1 < 1) ? (1 < 0) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((1 < 1) ? (1 < 1) ? 1 : 2 : 3 == 3)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/expressions/Triple.ycp"
    y2milestone ("true");
    return 1;
}
----------------------------------------------------------------------
[YCP] tests/expressions/Triple.ycp:44 true
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/expressions/Triple.ycp"
    y2milestone ("false");
    return 0;
}
----------------------------------------------------------------------
[YCP] tests/expressions/Triple.ycp:45 false
[Scanner] tests/expressions/Triple.ycp:46 Error: Expression before '?' must be boolean
[Scanner] tests/expressions/Triple.ycp:46 Error: Expected 'boolean', seen 'integer'.
Parsed:
----------------------------------------------------------------------
YError
----------------------------------------------------------------------
[Interpreter] tests/expressions/Triple.ycp:46 Expression before '?' must be boolean
