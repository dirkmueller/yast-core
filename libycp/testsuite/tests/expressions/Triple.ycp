# ---------------------------------------------------------
#
#  Filename:    Triple.ycp
#
#  Purpose:     test cases for YCP ternal operator ?:
#
#  Creator:     dan@suse.cz
#
#  Maintainer:  dan@suse.cz
#
# --------------------------------------------------------

# triple operator

(((size ([]) > 0) ? true : false) == false)
(contains ([1, 2, 3], 5) ? "yes" : "no" == "no")
((size ([]) > 0) ? 1 : -1 == -1)

((true?true?1:2:3) == 1)
((true?false?1:2:3) == 2)
((false?true?1:2:3) == 3)
((false?false?1:2:3) == 3)

((0<0?0<0?1:2:3) == 3)
((0<0?0<1?1:2:3) == 3)
((0<0?1<0?1:2:3) == 3)
((0<0?1<1?1:2:3) == 3)

((0<1?0<0?1:2:3) == 2)
((0<1?0<1?1:2:3) == 1)
((0<1?1<0?1:2:3) == 2)
((0<1?1<1?1:2:3) == 2)

((1<0?0<0?1:2:3) == 3)
((1<0?0<1?1:2:3) == 3)
((1<0?1<0?1:2:3) == 3)
((1<0?1<1?1:2:3) == 3)

((1<1?0<0?1:2:3) == 3)
((1<1?0<1?1:2:3) == 3)
((1<1?1<0?1:2:3) == 3)
((1<1?1<1?1:2:3) == 3)

(true?{y2milestone("true"); return 1;}:{y2milestone("false"); return 0;})
(false?{y2milestone("true"); return 1;}:{y2milestone("false"); return 0;})
(1?y2milestone("true"):y2milestone("false"))

