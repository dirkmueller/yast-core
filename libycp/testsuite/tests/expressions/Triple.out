(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(1)
(0)
